### PR TITLE
feat(TypeScript): Update types

### PR DESCRIPTION
**Pull request by ExE-Boss**
review?(@crutchcorn, @Elchi3, @vinyldarkscratch): This no longer depends on #3737 and can be merged as‑is.

### CODE REVIEW

File: test/test-versions.js [hunk 1]
```diff
@@ -106,10 +128,13 @@ function testVersions(dataFilename) {
...
 
+  /**
+   * @param {Identifier} data
+   */
   function findSupport(data) {
     for (const prop in data) {
-      if (prop === 'support') {
-        checkVersions(data[prop]);
+      if (prop === '__compat' && data[prop].support) {
+        checkVersions(data[prop].support);
```
Comment (ExE-Boss) [inline test/test-versions.js]: This operation was previously type unsafe, and could lead to a potential crash if there was a feature in BCD named `support`.

**Comment by ExE-Boss** [comment]
review?(@vinyldarkscratch): This&nbsp;no&nbsp;longer depends&nbsp;on&nbsp;#3737, so&nbsp;it&nbsp;can be&nbsp;merged as&#x2011;is.

**P.S.:** Can&nbsp;you please remove the&nbsp;<kbd>not&nbsp;ready</kbd>&nbsp;label?

File: types.d.ts [hunk 2]
```diff
@@ -1,8 +1,6 @@
 /* Any copyright is dedicated to the Public Domain.
  * http://creativecommons.org/publicdomain/zero/1.0/ */
 
-export as namespace bcd;
```
Comment (ExE-Boss) [inline types.d.ts]: I previously misunderstood what `export as namespace` meant.

I now know that this means export as a UMD global, which doesn’t exist.

**Comment by vinyldarkscratch** [comment]
I'm not entirely certain what problem this fixes (and to be honest, I'm not experienced with TypeScript).  From the basic tests to make sure the data still works as expected, I've found no issues.  Is there a specific issue this resolves, or is it more of a maintenance update?

**Comment by ExE-Boss** [comment]
This is&nbsp;more of&nbsp;a&nbsp;maintenance update so&nbsp;that the&nbsp;`types.d.ts` file reflects the&nbsp;schema.

**Pull request merged**